pkgs/applications/networking/cluster/waagent/default.nix

```diff
@@ -29,44 +22,33 @@ python39.pkgs.buildPythonPackage rec {
     sha256 = "0fvjanvsz1zyzhbjr2alq5fnld43mdd776r2qid5jy5glzv0xbhf";
   };
   patches = [
-    # Suppress the following error when waagent try to configure sshd:
+    # Suppress the following error when waagent tries to configure sshd:
     # Read-only file system: '/etc/ssh/sshd_config'
     ./dont-configure-sshd.patch
+
+    ./udev-set-mode-with-MODE.patch
   ];
   doCheck = false;
 
-  buildInputs = with python39.pkgs; [ distro ];
-  runtimeDeps = [
-    findutils
-    gnugrep
-    gnused
-    iproute2
-    iptables
-    nettools # for hostname
-    openssh
-    openssl
-    parted
-    procps # for pidof
-    shadow # for useradd, usermod
-    util-linux # for (u)mount, fdisk, sfdisk, mkswap
-  ];
+  propagatedBuildInputs = [ python.pkgs.distro ];
 
-  fixupPhase = ''
-     mkdir -p $out/bin/
-     WAAGENT=$(find $out -name waagent | grep sbin)
-     cp $WAAGENT $out/bin/waagent
-     wrapProgram "$out/bin/waagent" \
-         --prefix PYTHONPATH : $PYTHONPATH \
-         --prefix PATH : "${makeBinPath runtimeDeps}"
-     patchShebangs --build "$out/bin/"
+  # The binary entrypoint and udev rules are placed to the wrong place.
+  # Move them to their default location.
+  preFixup = ''
+    mv $out/${python.sitePackages}/usr/sbin $out/bin
```

Comment:
It might be waagent tries to re-execute itself with the python interpreter and fails on the bash script. I also need to take a look into that.

```
Dec 15 11:54:41 mymachine waagent[2039]:     PATH=${PATH:+':'$PATH':'}
Dec 15 11:54:41 mymachine waagent[2039]:          ^
Dec 15 11:54:41 mymachine waagent[2039]: SyntaxError: invalid syntax
Dec 15 11:54:42 mymachine waagent[2046]:   File "/nix/store/pallmwamaz2fmp515nfj98df6jdhfkc1-waagent-2.8.0.11/bin/waagent", line 2
Dec 15 11:54:42 mymachine waagent[2046]:     PATH=${PATH:+':'$PATH':'}
Dec 15 11:54:42 mymachine waagent[2046]:          ^
```

Reply:
Indeed waagent tries to re-execute itself, using argv0, which doesn't work with the way our wrapper works.

I replaced the move to $out/bin with a custom `makeWrapper` call, and disabled the python default binary wrapping procedures.